src/store.ts

```diff
@@ -93,18 +99,24 @@ export const initializeFirestoreListeners = (onLoad: () => void): (() => void) =
 
   let userNameInitialLoadFinished = false;
   let onboardingDataInitialLoadFinished = false;
+  let semestersInitialLoadFinished = false;
   let toggleableRequirementChoiceInitialLoadFinished = false;
   let subjectColorInitialLoadFinished = false;
   let uniqueIncrementerInitialLoadFinished = false;
 
+  let emitted = false;
+
   const emitOnLoadWhenLoaded = (): void => {
     if (
       userNameInitialLoadFinished &&
       onboardingDataInitialLoadFinished &&
+      semestersInitialLoadFinished &&
       toggleableRequirementChoiceInitialLoadFinished &&
       subjectColorInitialLoadFinished &&
-      uniqueIncrementerInitialLoadFinished
+      uniqueIncrementerInitialLoadFinished &&
+      !emitted
     ) {
+      emitted = true;
```

Comment:
lack of emitted is a bug. Without this flag set to true after first load emit, it will keep calling onLoad whenever there is an update in any listeners.